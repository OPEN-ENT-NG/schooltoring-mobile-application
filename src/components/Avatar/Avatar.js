import React from "react";
import { Image, View } from "react-native";
import axios from "axios";

import PropTypes from "prop-types";

const Avatar = props => (
  <View style={props.style}>
    <Image
      style={{
        height: props.size,
        width: props.size,
        borderRadius: props.size / 2,
        overflow: "hidden"
      }}
      source={
        props.src && props.src.indexOf("no-avatar.jpg") === -1
          ? {
              uri: `${props.src}`,
              headers: {
                Authorization: axios.defaults.headers.common["Authorization"]
              }
            }
          : require("../../assets/img/no-avatar.png")
      }
    />
  </View>
);

Avatar.propTypes = {
  size: PropTypes.number.isRequired,
  style: PropTypes.object,
  src: PropTypes.string
};

export default Avatar;
