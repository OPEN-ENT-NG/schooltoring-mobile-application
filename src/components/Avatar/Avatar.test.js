import React from "react";
import { shallow } from "enzyme";

import Avatar from "./Avatar";

import stateMocked from "../../jest/state.json";

describe("Render", () => {
  test("Avatar should render without crashing", () => {
    const rendered = shallow(<Avatar size={30} />);
    expect(rendered).toMatchSnapshot();
  });

  test("Avatar should render with given source and size", () => {
    const rendered = shallow(<Avatar size={30} src={stateMocked.testAvatar} />);
    expect(rendered).toMatchSnapshot();
  });

  test("An error should be logged if no size given", () => {
    const PropTypes = jest.spyOn(global.console, "error");
    const rendered = shallow(<Avatar />);
    expect(rendered).toMatchSnapshot();
    expect(PropTypes).toHaveBeenCalled();
  });
});
