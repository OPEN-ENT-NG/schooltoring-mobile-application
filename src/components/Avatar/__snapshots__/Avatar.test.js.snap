// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render An error should be logged if no size given 1`] = `
<View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/img/no-avatar.png",
      }
    }
    style={
      Object {
        "borderRadius": NaN,
        "height": undefined,
        "overflow": "hidden",
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`Render Avatar should render with given source and size 1`] = `
<View>
  <Image
    source={
      Object {
        "headers": Object {
          "Authorization": undefined,
        },
        "uri": "https://www.planetminecraft.com/files/resource_media/screenshot/1231/Megaman8bit_3126288_thumb.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 15,
        "height": 30,
        "overflow": "hidden",
        "width": 30,
      }
    }
  />
</View>
`;

exports[`Render Avatar should render without crashing 1`] = `
<View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/img/no-avatar.png",
      }
    }
    style={
      Object {
        "borderRadius": 15,
        "height": 30,
        "overflow": "hidden",
        "width": 30,
      }
    }
  />
</View>
`;
